### PR TITLE
Feature: skip-temp-tag

### PR DESCRIPTION
**Pull request by noherczeg**
I have no idea if this is a proper solution, therefore I'd like to have some guidance on how to properly test this.

### CODE REVIEW

**Comment by noherczeg** [comment]
In response to #293

File: src/commands/PublishCommand.js [hunk 1]
```diff
@@ -361,29 +362,35 @@ export default class PublishCommand extends Command {
...
-            this.logger.error("Attempting to retry publishing " + pkg.name + "...", err);
-            run(cb);
-          } else {
-            this.logger.error("Ran out of retries while publishing " + pkg.name, err);
-            cb(err);
-          }
-        });
+        if (skipTempTag) {
+          this.progressBar.tick(pkg.name);
+          this.execScript(pkg, "postpublish");
```
Comment (gigabo) [inline src/commands/PublishCommand.js]: Where is the actual publish happening with `--skip-temp-tag`?

**Comment by noherczeg** [comment]
So the idea is that
If we ignore temp tagging then:
- `npmPublishAsPrerelease()` should immediately set the proper tag value
- and `npmUpdateAsLatest()` should skip the `updateTag()` procedure which handles temp tag removal, and addition of the proper tags

If we did not ignore temp tagging then:
- `npmPublishAsPrerelease()` will publish with the "lerna-temp" tag
- while `npmUpdateAsLatest()` will trigger the `updateTag()` procedure...

**Comment by noherczeg** [comment]
I also introduced the `getDistTag()` method which might be an overkill, so if you guys feel like it I could remove it. I made it because as it stands in the code I kind of needed something like that in multiple places.

File: src/commands/PublishCommand.js [hunk 2]
```diff
@@ -394,6 +397,7 @@ export default class PublishCommand extends Command {
   }
 
   npmUpdateAsLatest(callback) {
+    const {skipTempTag} = this.getOptions();
```
Comment (evocateur) [inline src/commands/PublishCommand.js]: If we've skipped temp-tagging, why not just avoid the `runParallelBatches` call entirely?

```
const {skipTempTag} = this.getOptions();

if (skipTempTag) {
  return callback();
}

this.progressBar.init(...)
```

File: src/commands/PublishCommand.js [hunk 3]
```diff
@@ -433,4 +429,26 @@ export default class PublishCommand extends Command {
...
+    } else if (this.flags.canary) {
+      NpmUtilities.addDistTag(pkg.name, pkg.version, distTag, this.npmRegistry);
+    } else {
+      NpmUtilities.addDistTag(pkg.name, this.updatesVersions[pkg.name], distTag, this.npmRegistry);
+    }
+  }
+
+  getDistTag() {
+    const {npmTag, canary} = this.getOptions();
+    return npmTag ? npmTag : canary ? "canary" : "latest";
```
Comment (evocateur) [inline src/commands/PublishCommand.js]: I really dislike nested ternaries, especially all on one line. Wouldn't boolean defaults work here?

```
return npmTag || (canary && "canary") || "latest";
```

File: src/commands/PublishCommand.js [hunk 4]
```diff
@@ -349,6 +349,9 @@ export default class PublishCommand extends Command {
   }
 
   npmPublishAsPrerelease(callback) {
+    const {skipTempTag} = this.getOptions();
+    // if we skip temp tags we should tag with the proper value immediately therefore no updates will be needed
+    const tag = !skipTempTag ? "lerna-temp" : this.getDistTag();
```
Comment (evocateur) [inline src/commands/PublishCommand.js]: Let's invert the ternary (`const tag = skipTempTag ? this.getDistTag() : "lerna-temp";`) and make sure there's a newline between this assignment and the next `forEach` block.

File: src/commands/PublishCommand.js (hunk 2 above)
Comment (noherczeg) [inline src/commands/PublishCommand.js]: Yeah, I missed that. Corrected int #0c6ac7e

**Comment by evocateur** [comment]
LGTM, thanks for your responsiveness @noherczeg!

If @gigabo has no objections, I will merge this tomorrow.

**Pull request merged**